kura/examples/test/org.eclipse.kura.example.wire.bool.multiport.provider.test/src/main/java/org/eclipse/kura/example/wire/bool/multiport/provider/test/BoolLogicTest.java

```diff
@@ -0,0 +1,473 @@
+                    && BOOLEAN_OPERATION_DEFAULT.equals(props.get(BOOLEAN_OPERATION));
+        }
+        return false;
+    }
+
+    @Test
+    public void testAndOperation() throws Exception {
+        logger.info("### TESTING AND OPERATION ###");
+        CompletableFuture<WireEnvelope> out0Recfuture = outReceiver.nextEnvelope();
+        in0emitter.emit(new WireRecord(mapWithTrue));
+        in1emitter.emit(new WireRecord(mapWithTrue));
+
+        try {
+            WireRecord receivedRecord = out0Recfuture.get(1, TimeUnit.SECONDS).getRecords().get(0);
+            assertTrue((boolean) receivedRecord.getProperties().get(RESULT_NAME_DEFAULT).getValue());
+
+        } catch (TimeoutException e) {
+            fail("Timeout waiting for envelope");
+        }
+        in0emitter.emit(new WireRecord(mapWithTrue));
+        CompletableFuture<WireEnvelope> out1Recfuture = outReceiver.nextEnvelope();
+        in1emitter.emit(new WireRecord(mapWithFalse));
+        try {
+            WireRecord receivedRecord = out1Recfuture.get(1, TimeUnit.SECONDS).getRecords().get(0);
+            assertFalse((boolean) receivedRecord.getProperties().get(RESULT_NAME_DEFAULT).getValue());
+
+        } catch (TimeoutException e) {
+            fail("Timeout waiting for envelope");
+        }
+
+    }
+
+    @Test
+    public void testXorOperation() throws Exception {
+        logger.info("### TESTING XOR OPERATION ###");
+        Map<String, Object> props = new HashMap<>();
+        props.put(BOOLEAN_OPERATION, "XOR");
+
+        try {
+            WireTestUtil.updateWireComponentConfiguration(configurationService, UNDER_TEST_PID, props).get(30,
+                    TimeUnit.SECONDS);
+        } catch (InterruptedException | TimeoutException e) {
+            logger.error("Test error", e);
+            throw e;
+        }
+
+        CompletableFuture<WireEnvelope> out0Recfuture = outReceiver.nextEnvelope();
+        in0emitter.emit(new WireRecord(mapWithTrue));
+        in1emitter.emit(new WireRecord(mapWithTrue));
+
+        try {
+            WireRecord receivedRecord = out0Recfuture.get(1, TimeUnit.SECONDS).getRecords().get(0);
+            assertFalse((boolean) receivedRecord.getProperties().get(RESULT_NAME_DEFAULT).getValue());
+
+        } catch (TimeoutException e) {
+            fail("Timeout waiting for envelope");
+        }
+        in0emitter.emit(new WireRecord(mapWithTrue));
+        CompletableFuture<WireEnvelope> out1Recfuture = outReceiver.nextEnvelope();
+        in1emitter.emit(new WireRecord(mapWithFalse));
+        try {
+            WireRecord receivedRecord = out1Recfuture.get(1, TimeUnit.SECONDS).getRecords().get(0);
+            assertTrue((boolean) receivedRecord.getProperties().get(RESULT_NAME_DEFAULT).getValue());
+
+        } catch (TimeoutException e) {
+            fail("Timeout waiting for envelope");
+        }
+    }
+
+    @Test
+    public void testOrOperation() throws Exception {
```

Comment:
this test would succeed if there was a bug and a XOR would be calculated. A suggestion would be to test the whole truth table for all operators or selecting the inputs such that the result cannot be the result of another operator

Reply:
implemented all truth tables.